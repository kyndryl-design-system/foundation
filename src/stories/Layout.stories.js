import { html } from 'lit';

export default {
  title: 'Foundation/Layout',

  decorators: [
    (story) =>
      html`
        <style>
          .layout-examples .aspect-ratio-box {
            width: 500px;
            background: var(--kd-color-background-ui-medium);
            display: flex;
            align-items: center;
            justify-content: center;
          }

          .layout-examples .object-fit-box {
            max-width: 500px;
            border: 1px solid var(--kd-color-background-secondary);
          }
        </style>
        ${story()}
      `,
  ],
};

export const AspectRatio = {
  argTypes: {
    aspectRatio: {
      options: ['16x9', '2x1', '4x3', '1x1', '1x2', '3x4'],
      control: { type: 'select' },
      description: 'Change aspect ratio.',
      table: {
        type: { summary: 'select' },
        defaultValue: { summary: '16x9' },
      },
    },
  },

  args: {
    aspectRatio: '16x9',
  },

  render: (args) => {
    const aspectRatioClass = 'kd-layout__aspect-ratio--' + args.aspectRatio;
    return html`
      <div class="layout-examples">
        <h3>Aspect Ratio</h3>
        <div class="aspect-ratio-box  kd-type--display-02 ${aspectRatioClass}">
          ${args.aspectRatio}
        </div>
      </div>
    `;
  },
};

export const ObjectFit = {
  argTypes: {
    objectFit: {
      options: ['cover', 'contain'],
      control: { type: 'select' },
      description: 'Change object fit.',
      table: {
        type: { summary: 'select' },
        defaultValue: { summary: 'cover' },
      },
    },
  },

  args: {
    objectFit: 'cover',
  },

  render: (args) => {
    const objectFitClass = 'kd-layout__object-fit--' + args.objectFit;

    return html`
      <div class="layout-examples">
        <h3>Object Fit</h3>
        <p>A 16x9 image in a 4x3 container</p>
        <div class="object-fit-box kd-layout__aspect-ratio--4x3">
          <img
            class="${objectFitClass}"
            src="https://s7d1.scene7.com/is/image/kyndryl/ls_windingriver_16x9?qlt=85&wid=1200&ts=1650983065999&dpr=off"
          />
        </div>
      </div>
    `;
  },
};
