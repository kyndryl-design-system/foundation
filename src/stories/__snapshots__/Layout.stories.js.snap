// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foundation/Layout AspectRatio smoke-test 1`] = `
<style>
  .layout-examples .aspect-ratio-box {
            width: 500px;
            background: var(--kd-color-background-ui-subtle);
            display: flex;
            align-items: center;
            justify-content: center;
          }

          .layout-examples .object-fit-box {
            max-width: 500px;
            border: 1px solid var(--kd-color-background-secondary);
          }
</style>
<div class="layout-examples">
  <h3>
    Aspect Ratio
  </h3>
  <div class="aspect-ratio-box  kd-type--display-02 kd-layout__aspect-ratio--16x9">
    16x9
  </div>
</div>
`;

exports[`Foundation/Layout ObjectFit smoke-test 1`] = `
<style>
  .layout-examples .aspect-ratio-box {
            width: 500px;
            background: var(--kd-color-background-ui-subtle);
            display: flex;
            align-items: center;
            justify-content: center;
          }

          .layout-examples .object-fit-box {
            max-width: 500px;
            border: 1px solid var(--kd-color-background-secondary);
          }
</style>
<div class="layout-examples">
  <h3>
    Object Fit
  </h3>
  <p>
    A 16x9 image in a 4x3 container
  </p>
  <div class="object-fit-box kd-layout__aspect-ratio--4x3">
    <img src="https://s7d1.scene7.com/is/image/kyndryl/ls_windingriver_16x9?qlt=85&amp;wid=1200&amp;ts=1650983065999&amp;dpr=off"
         alt
         class="kd-layout__object-fit--cover"
    >
  </div>
</div>
`;
